cmake/RunTests.cmake

```diff
@@ -64,6 +64,14 @@ endif()
 
 set(ENV{SYSTEM_NAME} ${CMAKE_HOST_SYSTEM_NAME})  # used by test/helpers.lua.
 
+# Some tests may fail on fish shell #24941 #6172
+# In general $SHELL value can be pretty arbitrary and it seems like any
+# non-POSIX-compliant shell could cause some issues during a test run.
+#
+# Unsetting $SHELL makes neovim fall back to default value for 'shell' option.
```

Comment:
I don't see why we would want that. We set SHELL=sh in /test/old/ so why not use the same approach here?

Reply:
If we want to replicate what's being done in /test/old then we should set `$SHELL` to `sh` here and then set it to an empty string in `clear` only for Windows, right?

Reply:
Let's keep the decision in one place (either RunTests or clear()). This should work:

```
if(NOT WIN32)
  set(ENV{SHELL} sh)
endif()
```

Reply:
I updated this part. Seems to work just as well. I also changed comment a bit.